src/afl-fuzz-init.c

```diff
@@ -297,29 +297,27 @@ void setup_post(afl_state_t *afl) {
   dh = dlopen(fn, RTLD_NOW);
   if (!dh) { FATAL("%s", dlerror()); }
 
-  afl->post_handler = dlsym(dh, "afl_postprocess");
-  if (!afl->post_handler) { FATAL("Symbol 'afl_postprocess' not found."); }
-  afl->post_init = dlsym(dh, "afl_postprocess_init");
-  if (!afl->post_init) { FATAL("Symbol 'afl_postprocess_init' not found."); }
-  afl->post_deinit = dlsym(dh, "afl_postprocess_deinit");
-  if (!afl->post_deinit) {
+  struct custom_mutator * mutator;
+  mutator = ck_alloc(sizeof(struct custom_mutator));
+  memset(mutator, 0, sizeof(struct custom_mutator));
+
+  mutator->afl_custom_post_process = dlsym(dh, "afl_postprocess");
+  if (!mutator->afl_custom_post_process) { FATAL("Symbol 'afl_postprocess' not found."); }
+  mutator->afl_custom_init = dlsym(dh, "afl_postprocess_init");
```

Comment:
This should no longer be needed, the init is the same as the custom mutator initialization, right?

Reply:
it is for format of the post lib feature and that is what is name was/is.
it should not break what people are already using.
or what do you mean?

Reply:
`afl_postprocess_init` and `afl_custom_mutator_init` have exactly the same use. It makes no sense to support both/is error prone.

Reply:
Same for deinit.
IMHO after merge, people will have to rename the function and that's that.

Reply:
so you want to break the stuff people are using?
the afl_postprocess_init is what is in the post library, and we removed that extra code and handle as if it would be a custom mutator.

Reply:
so far we have the policy not to break what vanilla afl does. removing the post lib feature would change that. I am all for that, but for a future release where we also change defaults and remove unnecessary features (as we collect in the two issues). so breaking changes in one big release rather than every other release something might break. or do you think that is not a good approach?

Reply:
`afl_postprocess_init` is already AFL++ stuff. I've added that for globals. Nobody complained so far.
Also we drop a function in the custom mutator, so why not do it right?

Reply:
that is not correct.
```
google-afl # grep afl_post  experimental/post_library/post_library.so.c 
const unsigned char* afl_postprocess(const unsigned char* in_buf,
```
it is in vanilla afl

Reply:
oh sorry, you meant the _init/_deinit. you are right, that has to be fixed. doesnt exist in vanille, so it must be optional now